pkg/controllers/node/ipam.go

```diff
@@ -359,18 +636,57 @@ func (c *NodeController) kubernetesNodeForCalico(cnode string) (string, error) {
 	}
 
 	// Try to pull the k8s name from the retrieved Calico node object. If there is no match,
-	// this will return an empty string, correctly telling the calling code to ignore this allocation.
-	return getK8sNodeName(*calicoNode), nil
+	// this will return an ErrorNotKubernetes, indicating the node should be ignored.
+	return getK8sNodeName(*calicoNode)
 }
 
 func ordinalToIP(b *model.AllocationBlock, ord int) net.IP {
-	ip := b.CIDR.IP
-	var intVal *big.Int
-	if ip.To4() != nil {
-		intVal = big.NewInt(0).SetBytes(ip.To4())
-	} else {
-		intVal = big.NewInt(0).SetBytes(ip.To16())
-	}
-	sum := big.NewInt(0).Add(intVal, big.NewInt(int64(ord)))
-	return net.IP(sum.Bytes())
+	return b.OrdinalToIP(ord).IP
+}
+
+func (c *ipamController) syncDeleteEtcd() error {
```

Comment:
nit: Name of this is slightly misleading because while it only needs to be called for deleting any nodes in etcd mode, it really syncs up the nodes we get from the Calico API with the ones we get from the K8s API. Maybe something like `syncDeleteCalicoAndK8sNodes`?